packages/beeq/src/components/slider/bq-slider.tsx

```diff
@@ -302,15 +363,21 @@ export class BqSlider {
             part="track-area"
           />
           {/* PROGRESS AREA */}
-          <span
-            class="absolute top-1/2 h-1 w-1/2 -translate-y-1/2 rounded-xs bg-[--bq-slider--progress-color]"
-            ref={(elem) => (this.progressElem = elem)}
-            part="progress-area"
-          />
+          <div ref={(div: HTMLDivElement) => (this.inputDivElement = div)}>
+            <span
+              class="absolute top-1/2 h-1 w-1/2 -translate-y-1/2 rounded-xs bg-[--bq-slider--progress-color]"
+              ref={(elem) => (this.progressElem = elem)}
+              part="progress-area"
+            />
+          </div>
```

Comment:
I think there's no need to wrap the progress span elem for the purpose of getting its the width of the track area. You can rely on the `span[part="track-area"]` above for that.

```ts
private trackElem: HTMLSpanElement;

...

  // Get the width of the input container and the size of the slider thumb
  const inputWidth = this.trackElem.getBoundingClientRect().width;

...

{/* TRACK AREA */}
<span
  class="absolute top-1/2 h-1 w-full -translate-y-1/2 rounded-xs bg-[--bq-slider--trackarea-color]"
  ref={(elem) => (this.trackElem = elem)}
  part="track-area"
/>
{/* PROGRESS AREA */}
<span
  class="absolute top-1/2 h-1 w-1/2 -translate-y-1/2 rounded-xs bg-[--bq-slider--progress-color]"
  ref={(elem) => (this.progressElem = elem)}
  part="progress-area"
/>
```